src/google/protobuf/compiler/csharp/csharp_enum_field.cc

```diff
@@ -54,39 +53,38 @@ EnumFieldGenerator::EnumFieldGenerator(const FieldDescriptor* descriptor,
 EnumFieldGenerator::~EnumFieldGenerator() {
 }
 
-void EnumFieldGenerator::GenerateParsingCode(Writer* writer) {
-  writer->WriteLine("object unknown;");
-  writer->WriteLine("if(input.ReadEnum(ref result.$0$_, out unknown)) {", name());
+void EnumFieldGenerator::GenerateParsingCode(io::Printer* printer) {
+  printer->Print(variables_,
+		 "object unknown;\n"
+		 "if(input.ReadEnum(ref result.$name$_, out unknown)) {\n");
   if (SupportFieldPresence(descriptor_->file())) {
-    writer->WriteLine("  result.has$0$ = true;", property_name());
+    printer->Print(variables_,
+		   "  result.has$property_name$ = true;\n");
   }
-  writer->WriteLine("} else if(unknown is int) {");
+  printer->Print("} else if(unknown is int) {\n");
```

Comment:
C# uses to support generating files with CRLF vs LF line endings.
Printing "\n" everywhere makes supporting this option in the future much harder (if we decided to do that).
I am probably fine with this, just wanted to let you know what a consequence of this is.